homeassistant/components/supla/light.py

```diff
@@ -0,0 +1,175 @@
+"""Support for Supla lights - dimmer and rgb."""
+from __future__ import annotations
+
+import logging
+from math import ceil
+from pprint import pformat
+from typing import Any
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+import homeassistant.util.color as color_util
+
+from . import DOMAIN, SUPLA_COORDINATORS, SUPLA_SERVERS
+from .entity import SuplaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPLA_DIMMER = "DIMMER"
+SUPLA_RGBLIGHTING = "RGBLIGHTING"
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Supla lights."""
+    if discovery_info is None:
+        return
+
+    _LOGGER.debug("Discovery: %s", pformat(discovery_info))
+
+    entities: list[LightEntity] = []
+    for device in discovery_info.values():
+        device_name = device["function_name"]
+        server_name = device["server_name"]
+
+        if device_name == SUPLA_DIMMER:
+            entities.append(
+                SuplaDimmerEntity(
+                    device,
+                    hass.data[DOMAIN][SUPLA_SERVERS][server_name],
+                    hass.data[DOMAIN][SUPLA_COORDINATORS][server_name],
+                )
+            )
+
+        elif device_name == SUPLA_RGBLIGHTING:
+            entities.append(
+                SuplaRGBLightingEntity(
+                    device,
+                    hass.data[DOMAIN][SUPLA_SERVERS][server_name],
+                    hass.data[DOMAIN][SUPLA_COORDINATORS][server_name],
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class SuplaDimmerEntity(SuplaEntity, LightEntity):
+    """Representation of a Supla dimmable light."""
+
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the switch."""
+        if ATTR_BRIGHTNESS in kwargs:
+            # Convert from Home Assistant brightness (0-255) to (0-100)
+            brightness = min(100, ceil(kwargs[ATTR_BRIGHTNESS] * 100 / 255.0))
+            await self.async_action("SET_RGBW_PARAMETERS", brightness=brightness)
+            return
+        await self.async_action("TURN_ON")
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the switch."""
+        await self.async_action("TURN_OFF")
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if switch is on."""
+        if state := self.channel_data.get("state"):
+            return state["on"]
```

Comment:
Will this always be a bool, or do we need to convert it?
```suggestion
            return bool(state["on"])
```

Reply:
Based on the linked docs, this will always be a boolean. You can check `ChannelStateBrightness`, `ChannelStateColor` and `ChannelStateColorAndBrightness` response object definitions.